### PR TITLE
Dorosh/edx multi proctoring

### PR DESCRIPTION
**Pull request by dorosh**
@polesye 
description:
 it is possible to use on installation edx few proctoring backends
sandbox:
 http://master.raccoongang.com/
 http://studio.master.raccoongang.com/
 http://master.raccoongang.com:8020/ - web_assistent
 http://master.raccoongang.com:8030/ - web_assistent

### CODE REVIEW

File: common/lib/xmodule/xmodule/course_module.py [hunk 1]
```diff
@@ -34,6 +34,15 @@
 CATALOG_VISIBILITY_NONE = "none"
 
 
+def get_proctoring_list():
+    from django.conf import settings
```
Comment (maxsocl) [inline common/lib/xmodule/xmodule/course_module.py]: May be we need docstring here to describe exactly what happened?

Comment (polesye) [inline common/lib/xmodule/xmodule/course_module.py]: `from django.conf import settings`
XModules have to know nothing about `Django`

Comment (polesye) [inline common/lib/xmodule/xmodule/course_module.py]: Can we somehow use `@XBlock.needs('settings')` instead? Or add these settings to `Advanced Settings`?
https://github.com/edx/xblock-utils/wiki/Settings-and-theme-support

File: common/lib/xmodule/xmodule/course_module.py [hunk 2]
```diff
@@ -34,6 +34,15 @@
...
+    try:
+        proctoring_providers = settings.PROCTORING_BACKEND_PROVIDERS
+    except Exception:
```
Comment (polesye) [inline common/lib/xmodule/xmodule/course_module.py]: Be more precise. `AttributeError`?

File: lms/djangoapps/courseware/tests/test_module_render.py [hunk 3]
```diff
@@ -721,7 +721,7 @@ class TestProctoringRendering(SharedModuleStoreTestCase):
     @classmethod
     def setUpClass(cls):
         super(TestProctoringRendering, cls).setUpClass()
-        cls.course_key = ToyCourseFactory.create().id
+        cls.course_key = ToyCourseFactory.create(**{'proctoring_service': 'TEST_PROCTOR'}).id
```
Comment (polesye) [inline lms/djangoapps/courseware/tests/test_module_render.py]: `ToyCourseFactory.create(proctoring_service='TEST_PROCTOR')`

File: common/lib/xmodule/xmodule/course_module.py (hunk 2 above)
Comment (dorosh) [inline common/lib/xmodule/xmodule/course_module.py]: yes, AttributeError is better variant

File: lms/djangoapps/courseware/tests/test_module_render.py (hunk 3 above)
Comment (polesye) [inline lms/djangoapps/courseware/tests/test_module_render.py]: It've to be more explicit.

**Comment by auraz** [comment]
@Vvoous it is here for too long, can we close it?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Comment by raccoondeploy** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**